lib/pages/StudentDriverDetailsPage.dart

```diff
@@ -18,96 +20,103 @@ class StudentDriverDetailsPage extends StatefulWidget {
-
-          return FutureBuilder(
-            future: Future.wait([
-              studentDriversDao.queryStudents(args.id),
-              drivenTimeDao.queryAllDrivenTimesByStudentId(args.id)
-            ]),
-            builder: (context, snapshot) {
-              if (snapshot.connectionState != ConnectionState.done ||
-                  !snapshot.hasData) {
-                return Center(child: CircularProgressIndicator());
-              }
-
-              student = snapshot.data[0];
-              List<DrivenTime> drivenTimesList = snapshot.data[1].reversed
-                  .toList();
-
-              return ListView.builder(
-                itemCount: drivenTimesList.length + 1,
-                itemBuilder: (context, index) {
-                  if (index == 0) {
-                    return Padding(
-                      padding: EdgeInsets.symmetric(vertical: 15.0),
-                      child: Column(
-                        children: [
-                          Text(
-                            "${student.firstName} ${student.lastName}",
-                            style: TextStyle(fontSize: 32),
-                          ),
-                          Text(
-                            _getDrivenText(drivenTimesList),
-                            style: TextStyle(fontSize: 20.0),
-                          ),
-                          Text(_getFutureDrivesTimeText(drivenTimesList),
-                              style: TextStyle(fontSize: 20.0))
-                        ],
-                      ),
-                    );
-                  }
-
-                  return _lessonTile(drivenTimesList[index - 1]);
-                },
-              );
-            },
-          );
-        },
-      ),
-      floatingActionButton: FloatingActionButton(
-        onPressed: () async {
-          final drivenTimeDao =
-              (await StudentDriversDatabase.instance).drivenTimeDao;
-          var drivenTime = DrivenTime(
-              null, args.id, DateTime.now().add(Duration(minutes: 90)), 90);
-
-          await drivenTimeDao.insertTime(drivenTime);
-          if(totalTimeSoFar + 90 >= STUDENT_ALL_HOURS) {
-            final studentDao = (await StudentDriversDatabase.instance).studentDao;
-            student.allHours = true;
-            await studentDao.update(student);
-          }
-          setState(() {
-            //empty to requery db
-          });
-        },
+          return ListView.builder(
+            itemCount: drivenTimesList.length + 1,
+            itemBuilder: (context, index) {
+              if (index == 0) {
+                return _studentSummary();
+                }
+                return _lessonTile(drivenTimesList[index - 1]);
+              },
+            );
+          },
+        ),
+        floatingActionButton: FloatingActionButton(
+          onPressed: () async {
+            dynamic result = await Navigator.pushNamed(
```

Comment:
Jak dla mnie to lepiej dodawać godziny w ekranie dodawania godzin skoro jest to osobny page, co innego jakby był to tylko dialog

Reply:
No też o tym myślałem, ale skoro już utworzyłem całą infrastrukturę w `StudentDriverDetailsPage` no to nie chciałem znowu tego powtarzać, łączyć się z bazą, pobierać dao itd. Potraktowałem to jak okno dialogowe. Pobieramy info, wracamy i jest update. dzięki czemu nie powielamy kodu.

Swoją drogą na https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html przeczytałem, że nie powinno się tworzyć Future w FutureBuild tylko wcześniej, np initState, tam masz to uargumentowane. A widziałem, że tak zrobiłeś